internal/workflow.go

```diff
@@ -496,7 +505,7 @@ func AwaitWithTimeout(ctx Context, timeout time.Duration, condition func() bool)
 func (wc *workflowEnvironmentInterceptor) AwaitWithTimeout(ctx Context, timeout time.Duration, condition func() bool) (ok bool, err error) {
 	state := getState(ctx)
 	defer state.unblocked()
-	timer := NewTimer(ctx, timeout)
+	timer := NewTimerWithOptions(ctx, timeout, TimerOptions{"AwaitWithTimeout"})
```

Comment:
```suggestion
	timer := NewTimerWithOptions(ctx, timeout, TimerOptions{Summary: "AwaitWithTimeout"})
```

This is clearer to readers and a bit safer